routing/pathfind.go

```diff
@@ -323,7 +350,9 @@ func newRoute(amtToSend lnwire.MilliSatoshi, sourceVertex Vertex,
 		// As a sanity check, we ensure that the selected channel has
 		// enough capacity to forward the required amount which
 		// includes the fee dictated at each hop.
-		if nextHop.AmtToForward.ToSatoshis() > nextHop.Channel.Capacity {
+		if nextHop.Channel.Capacity != unknownCapacity &&
```

Comment:
Why's this check necessary? If it's a private channel and we don't yet know the capacity of it?

Reply:
This is necessary because the routing hint doesn't include the capacity of the channel, so it defaults to `0` when creating the `channeldb.ChannelEdgePolicy`.

Reply:
Since we probably can assume that the hints given to us have enough capacity to carry the payment, can't we just set `capacity = paymentAmt` (+ enough to account for fees) for the routing hints?

Reply:
Are there any drawbacks to the check included above? It was also useful in [`cc64a54`](https://github.com/lightningnetwork/lnd/pull/846/commits/cc64a54346161890d5c0bb000b89a38ea4251c05#diff-ef7018e489d1a522e140e134eda04fe9R565).

Reply:
Drawback as I see it is that another constant is introduced even not strictly necessary. If we always assumed the private channel had enough capacity, we wouldn't need the check in https://github.com/lightningnetwork/lnd/pull/846/commits/cc64a54346161890d5c0bb000b89a38ea4251c05#diff-ef7018e489d1a522e140e134eda04fe9R565.

Reply:
Fixed.